pkg/extension/workspace/example/runtime/getenvoy/runtime_test.go

```diff
@@ -17,166 +17,121 @@ package getenvoy_test
-
-		AfterEach(func() {
-			common.HomeDir = backupHomeDir
-		})
-
-		var tempHomeDir string
-
-		BeforeEach(func() {
-			dir, err := stdioutil.TempDir("", "getenvoy-home")
-			Expect(err).NotTo(HaveOccurred())
-			tempHomeDir = dir
-		})
-
-		AfterEach(func() {
-			if tempHomeDir != "" {
-				Expect(os.RemoveAll(tempHomeDir)).To(Succeed())
-			}
-		})
-
-		BeforeEach(func() {
-			common.HomeDir = tempHomeDir
-		})
-
-		envoyPath := func() string {
-			path, err := filepath.Abs("testdata/envoy/bin/envoy")
-			Expect(err).ToNot(HaveOccurred())
-			return path
-		}
-
-		var stdout *bytes.Buffer
-		var stderr *bytes.Buffer
-
-		BeforeEach(func() {
-			stdout = new(bytes.Buffer)
-			stderr = new(bytes.Buffer)
-		})
-
-		runContext := func(workspace model.Workspace, example model.Example) *runtime.RunContext {
-			return &runtime.RunContext{
-				Opts: runtime.RunOpts{
-					Workspace: workspace,
-					Example: runtime.ExampleOpts{
-						Name:    "default",
-						Example: example,
-					},
-					Extension: runtime.ExtensionOpts{
-						WasmFile: `/path/to/extension.wasm`,
-						Config: model.File{
-							Source:  "/path/to/config",
-							Content: []byte(`{"key2":"value2"}`),
-						},
-					},
-					Envoy: runtime.EnvoyOpts{
-						Path: envoyPath(),
-					},
-				},
-				IO: ioutil.StdStreams{
-					Out: stdout,
-					Err: stderr,
+// relativeWorkspaceDir points to a usable pre-initialized workspace
+const relativeWorkspaceDir = "../configdir/testdata/workspace1"
+
+func TestRuntimeRun(t *testing.T) {
+	workspace, err := workspaces.GetWorkspaceAt(relativeWorkspaceDir)
+	require.NoError(t, err, `expected no error getting workspace from directory %s`, relativeWorkspaceDir)
+
+	example, err := workspace.GetExample("default")
+	require.NoError(t, err, `expected no error getting example from workspace %s`, workspace)
+
+	fakeEnvoyPath, tearDown := setupFakeEnvoy(t)
+	defer tearDown()
+
+	// Create and run a new context that will invoke a fake envoy script
+	ctx, stdout, stderr := runContext(workspace, example, fakeEnvoyPath)
+	err = NewRuntime().Run(ctx)
+	require.NoError(t, err, `expected no error running running [%v]`, ctx)
+
+	// The working directory of envoy is a temp directory not controlled by this test, so we have to parse it.
+	envoyWd := extension.ParseEnvoyWorkDirectory(stdout)
+
+	// Verify we executed the indicated envoy binary, and it captured the arguments we expected
+	expectedStdout := fmt.Sprintf(`envoy pwd: %s
+envoy bin: %s
+envoy args: -c %s/envoy.tmpl.yaml
+`, envoyWd, ctx.Opts.Envoy.Path, envoyWd)
+	require.Equal(t, expectedStdout, stdout.String(), `expected stdout running [%v]`, ctx)
+
+	// Verify we didn't accidentally combine the stderr of envoy into stdout, or otherwise dropped it.
+	require.Equal(t, "envoy stderr\n", stderr.String(), `expected stderr running [%v]`, ctx)
+}
+
+func TestRuntimeRunFailsOnInvalidWorkspace(t *testing.T) {
+	invalidWorkspaceDir := extension.RequireAbsDir(t, "../configdir/testdata/workspace5")
```

Comment:
I would like `"../configdir/testdata/workspace5"` to be const like `workspace1` above :smile: